framework/db/oci/QueryBuilder.php

```diff
@@ -162,6 +162,27 @@ public function addForeignKey($name, $table, $columns, $refTable, $refColumns, $
     }
 
     /**
+     * For each return column a param will be added to $returnParams.
+     * They should have variables bound before executing the query.
+     * @inheritdoc
+     */
+    public function insertReturning($table, $columns, &$params, $returnColumns = null, &$returnParams = null)
+    {
+        $schema = $this->db->getSchema();
```

Comment:
Can't you just generate `$returnParams` inside this method?
As far as I can see it can be composed from column names.
This way method will not allow full flexibility of the Oracle, but will remain more generic.